Dependency updates, adds default light/dark mode support, adds additional default styles

armahillo opened the pull request:
I've been using this framework for a few years now and it's been my favorite boilerplate starting point. This PR represents the summation of numerous modifications I've made to it, over time, as well as some fixes to bring it current with changes to various JS dependencies (dart, sass, etc). I hope you find these changes useful and would be honored to have them integrated.

## Updates JS dependencies to bypass deprecations

[node-sass is dead; long live sass](https://sass-lang.com/blog/node-sass-is-end-of-life)

The new 'sass' command that replaces it has slightly different argument flags. I cross-referenced [node-sass](https://github.com/sass/node-sass?tab=readme-ov-file#command-line-interface) with [dart-sass](https://sass-lang.com/documentation/cli/dart-sass/#usage). These changes were necessary because of upstream deprecations that were blocking new development. 

Most of the changes were fairly straightforward.`-r` is now "many to many mode". Some of the embed flags changed. All npm `package.json` versions have been migrated to contemporary. I did not change `gulp` because I'm less familiar with it.

This also resulted in bumping the minor version from `1.5.3` to `1.6.0`

## Removes deprecation warnings by modernizing out of `@import` to `@use`

We were previously getting deprecation warnings for using deprecated functions (eg. `darken()`, `lighten()`). These deprecations were noted in the [sass-lang breaking changes doc](https://sass-lang.com/documentation/breaking-changes/color-functions/#single-channel-adjustment-functions).

The most direct solution is to use the [sass-migrator](https://sass-lang.com/documentation/cli/migrator/#installation), and then run it with:

```bash
sass-migrator module --migrate-deps main.scss
```

This updated all `@import` directives to be `@use` instead, and adapted all color functions to use the contemporary versions, eg. `adjust()` -> `color.adjust()`. It also changed all cross-file references to be explicit, eg. `$margins` -> `variables.$margins`.

## Color modes using `light-dark` are now implemented

As a person with mild photosensitivity, I find it [frustrating that more sites don't implement dark mode as an accessibility feature](https://armahillo.dev/fundamentals/2025/03/28/darkmode/). Previously I would address this through the use of CSS variables (as noted in the blog post), but this creates challenges with `primitive-ui`'s reliance on SASS variables, and that felt too out-of-scope to change.

With the recent widespread adoption of [`light-dark()`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/light-dark), adding support for dark mode becomes _significantly_ easier. This PR thus adds default support for responsive light/dark mode styling out-of-the-box, using that function. Extensive testing was done with the `dist/test.html` file, and additional test cases / checks were added to it, as well as directions on how to change system settings around lightness preference. Making in-page light/dark mode changes requires JS and is, frankly, unnecessary for accessibility reasons, especially since "auto" exists in major OSes. 

This also required the addition of a few new semantic color variables to handle these edge cases, and the "alternate-" prefix was changed to a "-alt" suffix. 

## Expands scope of `dist/test.html`

The previous `dist/test.html` kitchen sink demo was fairly comprehensive but lacked specific examples on some key features, particularly on the use of the `flex-row`, `flex-small`, `flex-large` classes, and their width helper classes. It also lacked several demonstrations of some different `input` fields (some of which are newer and may not have existed when Primitive UI was originally developed). Among these: `color`, `month`, `datetime-local`, etc. These were added to the `dist/test.html` to demonstrate how they can be displayed. I also expanded on the `.split-form` class since it seemed like the implementation was incomplete.

armahillo committed:
Updates JS dependencies to bypass deprecations

node-sass is dead; long live sass

https://sass-lang.com/blog/node-sass-is-end-of-life

The new 'sass' command that replaces it has slightly different argument
flags. I cross-referenced:

https://github.com/sass/node-sass?tab=readme-ov-file#command-line-interface

with

https://sass-lang.com/documentation/cli/dart-sass/#usage

Most of the changes were fairly straightforward. '-r' is now "many to
many mode". Some of the embed flags changed.

I did not check Gulp yet.

All npm package.json versions have been migrated to contemporary.

armahillo committed:
Removes deprecation warnings by modernizing out of @import to @use

We were previously getting deprecation warnings for using deprecated
functions (eg. darken(), lighten()). These deprecations were noted in:

https://sass-lang.com/documentation/breaking-changes/color-functions/#single-channel-adjustment-functions

The most direct solution is to use the sass-migrator

https://sass-lang.com/documentation/cli/migrator/#installation

And then run it with:

sass-migrator module --migrate-deps main.scss

This updated all @import directives to be @use instead, and adapted all
color functions to use the contemporary versions: color.adjust()

armahillo committed:
Minor bugfix on sass:build command. Rebuilds dist/main.min.css

armahillo committed:
Color modes using light-dark are now implemented

armahillo committed:
Adds additional demos to the test file

armahillo committed:
Final changes to kitchen sink and styles. Adds final inputs and adjusts colors. Fixes light-dark syntax issues

taniarascia commented:
Do the `light-dark` changes work with the docs? When I pull down the PR and build the docs, they look like this for me, defaulting to a few dark mode styles, but it doesn't look right:

<img width="956" height="893" alt="Screen Shot 2025-12-01 at 9 31 28 PM" src="https://github.com/user-attachments/assets/67593494-aff4-4562-938b-b6a03cf1daf1" />

armahillo commented:
@taniarascia Oh I didn't check the docs build -- Agreed that does not look right! Let me look into it

for clarification: is your system (or at least your browser) in "Dark" mode, I presume? the `light-dark` CSS function should be looking at system preference to determine which to display.

I was able to repro it -- I also found a bug in the javascript (switching back to "classic" from "eighties" doesn't work -- including a fix for that). Thanks for the catch!

armahillo committed:
Fixes example.html in dark mode, CSS and JS cleanup

This commit does:

- Moves some contents from _variables.scss that was being doubly-echoed
  in the final main.css compilation.
- Replaces /* comments with // in the SCSS so that some comments don't
  get conveyed to the final main.css compilation.
- Adds some styles to example.css and eighties.css so that they are both
  readable (with sufficient contrast) in light and dark mode
- Modernizes CSS of example.css and eighties.css to use nesting, now natively supported in regular CSS
- Removes some explicit styles from example.css that are already handled
  by the Primitive UI base CSS.
- Applies "alternate-background" class to top section and aside
- Replaces placeholder.it (defunct) with placehold.co (new)
- Removes rogue style override in h3 on demo page
- Corrects a jQuery selector to allow it to accommodate
  "eighties.css?version=2" (it previously was failing to switch back)
- Modified the scroll/click functions in the JS file to address two
  minor behaviors: double firing (on click) and overzealous scroll
handling.

armahillo commented:
@taniarascia OK please check out the latest on here. I did [several updates](https://github.com/taniarascia/primitive/pull/32/commits/415d7fe7c479ec3af8d12720f894a40c801cd9d6) which I think address the issues in the demo page in dark mode. I verified the display on both normal and eighties mode. I had to add a couple extra styles to "example.css" and "eighties.css" but I tried to use a light touch with that. 

There were a couple minor bugs I found in the JS that I was able to correct as well. 

LMK if you see anything else!

taniarascia commented:
This looks great! Thank you for your excellent contribution and ensuring Primitive continues to be useful.
